packages/sn-controls-react/src/fieldcontrols/FileName.tsx

```diff
@@ -23,8 +23,8 @@ export function FileName(props: ReactClientFieldSetting) {
   const [value, setValue] = useState(valueInitialState)
 
   const getExtension = () => {
-    if (props.content && props.content[props.settings.Name] && props.content[props.settings.Name].indexOf('.') > -1) {
-      return props.content[props.settings.Name].substr(props.content[props.settings.Name].lastIndexOf('.') + 1)
+    if (props.fieldValue && props.fieldValue.indexOf('.') > -1) {
```

Comment:
If you try to update the name itself the extension is added to it on every keyup

![image](https://user-images.githubusercontent.com/6968860/61533640-bea14100-aa2d-11e9-9838-4ea3980d8be3.png)

Reply:
fixed